rdkit/Chem/FeatMaps/FeatMaps.py

```diff
@@ -154,41 +153,43 @@ def GetFeatFeatScore(self, feat1, feat2, typeMatch=True):
 
     return score
 
-  def ScoreFeats(self, featsToScore, mapScoreVect=[], featsScoreVect=[], featsToFeatMapIdx=[]):
+  def ScoreFeats(self, featsToScore, mapScoreVect=None, featsScoreVect=None, featsToFeatMapIdx=None):
     nFeats = len(self._feats)
-    if mapScoreVect and len(mapScoreVect) != nFeats:
-      raise ValueError('if provided, len(mapScoreVect) should equal numFeats')
-    nToScore = len(featsToScore)
-    if featsScoreVect and len(featsScoreVect) != nToScore:
-      raise ValueError('if provided, len(featsScoreVect) should equal len(featsToScore)')
-    if featsToFeatMapIdx and len(featsToFeatMapIdx) != nToScore:
-      raise ValueError('if provided, len(featsToFeatMapIdx) should equal len(featsToScore)')
-
-    if mapScoreVect:
+    if mapScoreVect is not None:
+      if len(mapScoreVect) != nFeats:
+        raise ValueError('if provided, len(mapScoreVect) should equal numFeats')
       for i in range(nFeats):
         mapScoreVect[i] = 0.0
     else:
       mapScoreVect = [0.0] * nFeats
 
+    nToScore = len(featsToScore)
     if self.scoreMode == FeatMapScoreMode.Closest:
       defScore = 1000.0
-    else:
+    else: 
       defScore = 0.0
-    if featsScoreVect:
+      
+    if featsScoreVect is not None:
+      if len(featsScoreVect) != nToScore:
+        raise ValueError('if provided, len(featsScoreVect) should equal len(featsToScore)')
       for i in range(nToScore):
         featsScoreVect[i] = defScore
     else:
       featsScoreVect = [defScore] * nToScore
 
-    if not featsToFeatMapIdx:
+    if featsToFeatMapIdx is not None: # Initialize a 2D-empty array
+      if len(featsToFeatMapIdx) != nToScore:
+        raise ValueError('if provided, len(featsToFeatMapIdx) should equal len(featsToScore)')
+    else:
       featsToFeatMapIdx = [None] * nToScore
-
-    for i in range(nToScore):
-      if self.scoreMode != FeatMapScoreMode.All:
+    
+    if self.scoreMode != FeatMapScoreMode.All:
```

Comment:
I'm going to approve this PR anyway, but I think this change makes the code harder to read, not easier.
There is probably a small performance improvement for this form, but this is not in a time-sensitive piece of the code, so I think it's not worth it.

Reply:
Yes, I agreed. But the important is related to the default. As known, default variable as a list is hard to control. Switch to None will reduce memory leak

Reply:
I think @greglandrum was referring to moving the `if` statement out of the loop. 

Changing the default for the lists in the keyword arguments from `[]` to `None` makes sense and was a reasonable change.

Reply:
> I'm going to approve this PR anyway, but I think this change makes the code harder to read, not easier. There is probably a small performance improvement for this form, but this is not in a time-sensitive piece of the code, so I think it's not worth it.

OK, I got your idea, I will fix it now